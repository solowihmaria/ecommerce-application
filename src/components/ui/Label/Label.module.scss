@use '../../../assets/styles/constants.module' as c;

.label {
    display: inline-block;
    margin-bottom: 0.5rem;
    font-family: inherit;
    font-size: 1rem;
    font-weight: 500;
    line-height: 1.5;
    color: c.$color-text-primary;

    &.disabled {
        color: #999;
        cursor: not-allowed;
        color: c.$color-gray-500;
    }

    span[aria-hidden='true'] {
        margin-left: 0.25rem;
        color: c.$color-error;
    }
}
