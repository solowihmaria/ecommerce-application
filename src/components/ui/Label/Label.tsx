import styles from './Label.module.scss';
import clsx from 'clsx';
import type { LabelProps } from './Label.types';

export const Label = ({
    htmlFor,
    className = '',
    required,
    disabled,
    children,
    ...props
}: LabelProps) => {
    return (
        <label
            htmlFor={htmlFor}
            className={clsx(
                styles.label,
                className,
                disabled && styles.disabled,
                required && styles.required
            )}
            {...props}
        >
            {children}
            {required && <span aria-hidden="true">*</span>}
        </label>
    );
};
