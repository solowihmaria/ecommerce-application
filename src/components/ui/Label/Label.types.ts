import type { LabelHTMLAttributes } from 'react';

export interface LabelProps extends LabelHTMLAttributes<HTMLLabelElement> {
    htmlFor?: string;
    required?: boolean;
    disabled?: boolean;
}
