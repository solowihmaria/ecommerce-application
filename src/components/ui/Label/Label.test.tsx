import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Label } from './Label';
import type { LabelProps } from './Label.types';

describe('<Label/> component', () => {
    test('Render Label component without any properties', () => {
        render(<Label>label text</Label>);
        const label = screen.getByText(/label text/i);
        expect(label).toBeInTheDocument();
        expect(label).toMatchSnapshot();
    });

    test('Render Label component - optional field', () => {
        const properties: LabelProps = {
            htmlFor: 'someInput',
            children: 'LabelText',
            className: 'labelClass',
            required: false,
            disabled: false,
        };
        render(<Label {...properties} />);
        const label = screen.getByText(/labeltext/i);
        expect(label).toBeInTheDocument();
        expect(label).toHaveClass('labelClass');
        expect(label).toMatchSnapshot();
    });

    test('Render Label component - required field', () => {
        const properties: LabelProps = {
            htmlFor: 'someInput',
            children: 'LabelText',
            className: 'labelClass',
            required: true,
            disabled: false,
        };
        render(<Label {...properties} />);

        const label = screen.getByText(/labeltext/i);
        const requiredMark = screen.getByText(/\*/i);

        expect(label).toBeInTheDocument();
        expect(requiredMark).toBeInTheDocument();
        expect(label).toMatchSnapshot();
    });

    test('Render Label component - disabled', () => {
        const properties: LabelProps = {
            htmlFor: 'someInput',
            children: 'LabelText',
            className: 'labelClass',
            required: true,
            disabled: true,
        };
        render(<Label {...properties} />);
        const label = screen.getByText(/labeltext/i);
        expect(label).toBeInTheDocument();
        expect(label).toHaveClass('disabled');
        expect(label).toMatchSnapshot();
    });
});
