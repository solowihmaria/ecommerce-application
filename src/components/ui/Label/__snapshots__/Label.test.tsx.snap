// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> component Render Label component - disabled 1`] = `
<label
  class="label labelClass disabled required"
  for="someInput"
>
  LabelText
  <span
    aria-hidden="true"
  >
    *
  </span>
</label>
`;

exports[`<Label/> component Render Label component - optional field 1`] = `
<label
  class="label labelClass"
  for="someInput"
>
  LabelText
</label>
`;

exports[`<Label/> component Render Label component - required field 1`] = `
<label
  class="label labelClass required"
  for="someInput"
>
  LabelText
  <span
    aria-hidden="true"
  >
    *
  </span>
</label>
`;

exports[`<Label/> component Render Label component without any properties 1`] = `
<label
  class="label"
>
  label text
</label>
`;
